### PR TITLE
Add node configuration for requiring files to be images #9333

### LINKED ISSUE #9333: Add Image Datatype
Currently images are managed in arches using the file-list datatype. However, there is commonly data associated with images such as caption, credit, and location that currently must be recorded in separate nodes in a card. With an image datatype this information could be saved alongside the url, perhaps accompanied by other image information such as dimensions and pixel depth. This would reduce the number of nodes needed to track metadata about images. It would also simplify design around components such as the 'PhotoGalleryCard' or that are not intended for use with other file types.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Adds `imagesOnly` configuration to `file-list` nodes so that if set, the validator will reject anything that's not an image. This relies on Django's built-in image validation.

Adds fields for ~title, caption, description~ alt text, title, attribution, description to dropzone widget.

Adds validation to graph designer to prevent adding an images-only restriction after non-images already saved.

Adds data validation when saving files such that images-only nodes reject non-images and files without alt texts.

### Issues Solved
Closes #9333

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: Getty Conservation Institute
*   Found by: @chiatt
*   Tested by: @jacobtylerwalls
*   Designed by: @ 

#### Further comments
Do we need to add a max length for title, caption, description?

### CODE REVIEW

**Review by apeters — CHANGES_REQUESTED** [review]
Screenshot 2023-09-20 at 5 59 09 PM

I'm not sure what the difference between `title` and `caption` are.  Maybe only one or the other is needed.  Also the report never utilizes any of these values.  I would assume that the `alt` attribute of the image (and the file too), would use either the `title` or `caption`.  Not sure where the description shows up.  Maybe that's a text block next to the image or file.
We will probably need further discussion on that.

**Comment by jacobtylerwalls** [comment]
I’m starting to think we should call that second input “alt text” and just forget caption so that people are inspired to write good alt texts instead of good captions/bad alt texts.

If we were building something simple like a blog with captioned images, I guess a representative title for an image would be “Opening Day 2024” or “Benjamin Franklin, 17xx-18xx” with captions, respectively, “Better things to come?”, “Philadelphia’s most famous resident”, neither of which describe the image content in enough detail for an alt text. The alt texts would need to be something like “patrons queuing outside the XYZ Museum’s new location” and “a bust of Benjamin Franklin”. But Arches isn’t a blog, so maybe we can just have title, alt text, and a more verbose description field.

Some validation on the length of the alt text would be nice to prevent people from spamming screen readers with a wall of text.

File: arches/app/templates/views/components/datatypes/file-list.htm [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
+<div class="relative">
+    <div class="node-config-item">
+        <label class="col-xs-12 control-label widget-input-label">
+            <p class="control-label node-widget-label">
+                <span data-bind="text: $root.translations.maximumNumberOfFiles"></span>
+            </p>
+            <input class="form-control input-md widget-input widget-file" placeholder="1" type="number" data-bind="textInput: maxFiles"></input>
+        </label>
+    </div>
+    <div class="node-config-item">
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply arches/app/templates/views/components/datatypes/file-list.htm:12]: Nice catch. Turns out this is also a problem with the maximum number of files control just above it (double checked on 7.5)

**Comment by jacobtylerwalls** [comment]
> We will probably need further discussion on that.

Exactly yeah, I figured putting this up for review would stir some design discussion, broke that out into #10050.

File: arches/app/templates/views/components/datatypes/file-list.htm (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply arches/app/templates/views/components/datatypes/file-list.htm:12]: #10059

File: arches/app/templates/views/components/widgets/file.htm [hunk 2]
```diff
@@ -117,6 +117,41 @@
                             </div>
                         </div>
                     </div>
+                    <div class="row">
+                        <div class="col-sm-4">
+                            <div style="padding: 0px 4px;">
+                                <label for="title">Title</label>
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply arches/app/templates/views/components/widgets/file.htm]: Very helpful. [9f3d8be](https://github.com/archesproject/arches/pull/9934/commits/9f3d8be851b8a8fabea2e8448439fbbe640123e1)

**Comment by jacobtylerwalls** [comment]
@chiatt @apeters 

Updated with the changes we discussed yesterday:
- redesign metadata fields, add "?" icon and dropdown
- require alt texts (server-side) for image-only nodes
- localize string metadata

In another PR I'll look at #10059. If we want to show anything on the report I figure we should get more design input on #10050.

**Comment by jacobtylerwalls** [comment]
I keep finding more little nits with manipulating the widget (e.g. removing one of multiple files) while there's unsaved data. Unfortunately the complexity level is already higher than I'd like!

This is a lot closer, but there's still an issue with:
1. Upload file, save metadata
2. Edit metadata
Result: form not detected in dirty state, no save control

I'll take a look tomorrow.

**Comment by jacobtylerwalls** [comment]
Thanks for your patience @apeters, this is ready for another look.

**Review by apeters — CHANGES_REQUESTED** [review]
resaving a file, doesn't clear out dirty state.  See this: https://recordit.co/6Wr5ILUpab

**Review by apeters — APPROVED** [review]
looks good @jacobtylerwalls !

**Comment by jacobtylerwalls** [comment]
Thanks for the review!

**Pull request merged**